transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -47,6 +49,7 @@
             new IllegalStateException("Channel is unhealthy not offering it back to pool"),
             SimpleChannelPool.class, "releaseAndOffer(...)");
 
+    protected final ConcurrentMap<Channel, Channel> allChannels = PlatformDependent.newConcurrentHashMap();
```

Comment:
Why ? Just add it the `ChannelGroup` and call `ChannelGroup.close()` at the end. ChannelGroup is a Set.